tools/topology/topology1/platform/intel/intel-generic-hdmi.m4

```diff
@@ -0,0 +1,124 @@
+#
+# Topology for HDMI
+#
+
+include(`hda.m4')
+
+# variable that need to be defined in upper m4
+ifdef(`PLATFORM',`',`fatal_error(note: Need to define platform identifier for intel-generic-hdmi
+)')
+ifdef(`HDMI_PCM_ID_BASE',`',`fatal_error(note: Need to define HDMI pcm id for intel-generic-hdmi
+)')
+ifdef(`HDMI_PIPELINE_ID_BASE',`',`fatal_error(note: Need to define HDMI pipeline id for intel-generic-hdmi
+)')
+ifdef(`HDMI_DAI_LINK_ID_BASE',`',`fatal_error(note: Need to define HDMI dai id for intel-generic-hdmi
+)')
+
+ifelse(
+	PLATFORM, `cml', `define(`HDMI_DEV_NUM', 3)', 
+	PLATFORM, `jsl', `define(`HDMI_DEV_NUM', 3)', 
+	PLATFORM, `tgl', `define(`HDMI_DEV_NUM', 4)', 
+	PLATFORM, `adl', `define(`HDMI_DEV_NUM', 4)',
+	`fatal_error(note: Unknown platform to intel-generic-hdmi)')
+
+#
+# Define the pipelines
+#
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     period, priority, core,
+dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
+dnl     time_domain, sched_comp)
+
+`# Low Latency playback pipeline 'HDMI_PIPELINE_ID_BASE` on PCM 'HDMI_PCM_ID_BASE` using max 2 channels of s32le.'
+# Schedule 48 frames per 1000us deadline with priority 0 on core 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	HDMI_PCM_ID_BASE, HDMI_PIPELINE_ID_BASE, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+`# Low Latency playback pipeline 'eval(HDMI_PIPELINE_ID_BASE + 1)` on PCM 'eval(HDMI_PCM_ID_BASE + 1)` using max 2 channels of s32le.'
+# Schedule 48 frames per 1000us deadline with priority 0 on core 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	eval(HDMI_PCM_ID_BASE + 1), eval(HDMI_PIPELINE_ID_BASE + 1), 2, s32le,
```

Comment:
sorry, I'meant a separate M4 to define the pipeline IDs based on the base ID. Is this how it is done in other files ?

Reply:
I added a intel-boards.m4 to define the dai link ids which are supposed to match the same id in machine driver